src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,592 @@
+            }
+            GCText {
+                id: feedback
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: first_screen.visible ? '' :(items.toAlpha ?
+                          qsTr("Morse value: %1").arg(value) :
+                          qsTr('Alphabet/Numeric value: %1').arg(value))
+                onTextChanged: timer.start()
+                color: 'white'
+                fontSizeMode: Text.Fit
+                Rectangle {
+                    z: -1
+                    opacity: 0.8
+                    gradient: Gradient {
+                        GradientStop { position: 0.0; color: "#000" }
+                        GradientStop { position: 0.9; color: "#666" }
+                        GradientStop { position: 1.0; color: "#AAA" }
+                    }
+                    radius: 10
+                    border.color: 'black'
+                    border.width: 1
+                    anchors.centerIn: parent
+                    width: parent.width * 1.1
+                    height: parent.height
+                }
+                property string value: first_screen.visible ? '' : items.toAlpha ?
+                                           morseConverter.morse.trim() :
+                                           morseConverter.alpha ? morseConverter.alpha : ''
+            }
+
+            Timer {
+                id: timer
+                interval: 500
+                onTriggered: {
+                    if( feedback.value == items.questionValue ) {
+                        bonus.good('tux')
+                        textInput.clear()
+                    }
+
+                }
+            }
+            Item {
+                id:margin
+                width: 1
+                height: 20 * ApplicationInfo.ratio
+            }
+
+            GCText {
+                id: instruction
+
+                fontSize: largeSize
+                minimumPixelSize: 70
+                fontSizeMode: Text.Fit
+                horizontalAlignment: Text.AlignHCenter
+                anchors.horizontalCenter: parent.horizontalCenter
+                width: parent.width * 0.5
+                text: items.instruction
+                height: Text.Fit
+                wrapMode: TextEdit.WordWrap
+                color: 'black'
+
+            }
+
+
+        }
+
+        Rectangle {
+            id: morse_map
+            width: parent.width * 0.8
+            height: parent.height * 0.7
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.verticalCenter: parent.verticalCenter
+            visible: false
+            Grid {
+                    width: parent.width
+                    height: parent.height
+                    anchors.leftMargin: 50
+                    anchors.rightMargin: 10
+                    anchors.margins: 20
+                    columns: 5
+                    spacing: 20
+
+                    Text { text: " A = .-"; font.pixelSize: 40 }
```

Comment:
No, it's not what I have in mind for this.
1) You should never user Text but GCText (in GCompris ;)).
2) You should have something like in braille: https://github.com/gcompris/GCompris-qt/blob/master/src/activities/braille_alphabets/BrailleMap.qml#L50 which will automatically handle the resizing and positioning of the elements.
Instead of having the conversion tables in the morse2alpha and alpha2morse functions, you can have them outside, that would avoid you to duplicate them when needed.